src/pkcs11/interface.rs

```diff
@@ -6,6 +6,14 @@ include!("bindings.rs");
 // types that need different mutability than bindgen provides
 pub type CK_FUNCTION_LIST_PTR = *const CK_FUNCTION_LIST;
 pub type CK_FUNCTION_LIST_3_0_PTR = *const CK_FUNCTION_LIST_3_0;
+// this is wrongly converted on 32b architecture to too large value
+// which can not be represented in CK_ULONG. Docs says it can't have
+// any other value so enumerate them here:
+// https://doc.rust-lang.org/std/os/raw/type.c_ulong.html
+#[cfg(target_pointer_width = "64")]
+pub const CK_UNAVAILABLE_INFORMATION: CK_ULONG = u64::MAX;
+#[cfg(target_pointer_width = "32")]
+pub const CK_UNAVAILABLE_INFORMATION: CK_ULONG = u32::MAX;
```

Comment:
have you tried to just use CK_ULONG::MAX ?
Or maybe (-1 as CK_ULONG) ?

Reply:
To be clear, I think on windows unsigned long is 32 bit while pointer is 64 bit, which is why your test is not correct.

Reply:
Yes, the warning is gone, but the other issues are still present.